key_management/src/keystore.rs

```diff
@@ -131,3 +136,66 @@ impl KeyStore for MemKeyStore {
         self.key_info.remove(&key)
     }
 }
+
+/// KeyStore that persists data in KEYSTORE_LOCATION
+#[derive(Default, Clone, PartialEq, Debug, Eq)]
+pub struct PersistentKeyStore {
+    pub key_info: HashMap<String, KeyInfo>,
+    location: String,
+}
+
+impl PersistentKeyStore {
+    pub fn new(location: String) -> Self {
+        let file_op = File::open(&location);
+        match file_op {
+            Ok(file) => {
+                let reader = BufReader::new(file);
+                let data: HashMap<String, KeyInfo> = serde_json::from_reader(reader).unwrap();
+                Self {
+                    key_info: data,
+                    location,
+                }
+            }
+            Err(_) => Self {
```

Comment:
should maybe add some log or something to indicate the file doesn't exist (silently refreshing the keystore if any error)